File: languages/templates/python/checker.py
import io
from contextlib import redirect_stdout

def run_checker(testcases, solution):
    # results:
    # - exception
    # - isPassed
    # - failedTestcases:
    #      - id
    #      - input
    #      - expected
    #      - got
    #      - stdOut
    results = {
        "isPassed": True,
        "failedTestcases": [],
    }

    for testcase in testcases:
        testcase_input = testcase["input"]
        expected = testcase["expected"]
        result = {}

        try:
            stdout = io.StringIO()
            with redirect_stdout(stdout):
                # got = solution.FUNCTION(input)

            if got != expected:
                result["id"] = testcase["id"]
                result["input"] = testcase_input
                result["expected"] = expected
                result["got"] = got
                result["stdOut"] = stdout.getvalue()

                results["isPassed"] = False
                results["failedTestcases"].append(result)
        except Exception as e:
            results["exception"] = e
            results["isPassed"] = False
            break

    return results

File: languages/templates/go/checker.go
package main

func runChecker(testcases []*Testcase) *RunResult {
	runResult := new(RunResult)

	for _, testcase := range testcases {
		// input
		// expected
		// got

		if expected != got {

			runResult.FailedTestcases = append(runResult.FailedTestcases, &FailedTestcase{
				Id:       testcase.Id,
				Input:    testcase.Input,
				Expected: testcase.Expected,
				Got:      got,
			})
			runResult.IsPassed = false
		}
	}

	return runResult
}
